### ISSUE TITLE
Replace MATLAB's smart indent function

### ISSUE THREAD

**Issue by mriesch-tum** [davidvarga/MBeautifier#75]
First of all, thank you for this project! Until yesterday I was not aware that such a formatter exists and spent some time formatting things manually. This project really makes things easier!

**Problem:** The issues #51 and #74 already mention that it would be very nice to run the script without the MATLAB editor dependence. For automated tasks (pre-commit hooks, CI, ...) this is crucial as there might not be the possibility to run a GUI. For a large number of files this may be beneficial for the performance. Apparently, the only dependency is the smart indentation function which must be replaced. The replacement could also deal with indentation issues such as #52 .

**Approach 1:** Write a MATLAB function that can replace MATLAB's smart indent function. As announced in #74, I played around and ended up with a simple indentation code which I would like to share (for what it is worth).

**Approach 2:** Implement the same thing in Java, as the project is about to be rewritten. May I ask what the status of the Java version is?

**Comment by mriesch-tum** [comment]
I agree, a full parser is required. Also, I understand that your time budget is limited, no apologies required here :-) Is there some Java code available on GitHub already? Maybe there are some people who are willing to help.

As to pull request #76, consider it as quick-and-dirty fix to bridge the time until the Java version is ready. Maybe someone can extend or use it.

**Comment by florianschanda** [comment]
@mriesch-tum and @davidvarga I have managed to write a fully working MATLAB lexer and parser (https://github.com/florianschanda/miss_hit), and I plan to write down the formal grammar sometime soon.
